test/library/draft/arrow/ArrowRecord.chpl

```diff
@@ -0,0 +1,83 @@
+module ArrowRecord {
+  // As opposed to other modules with similar names to Chapel files, this one is not
+  // a transliteration of arrowrecord.c into Chapel.
+  // Rather it is the high-level module used to build a record batch and schema and tables 
+  // and stuff.
+  require "ArrayDecl.chpl", "ArrowRecordDecl.chpl";
+  private use ArrayDecl, ArrowRecordDecl;
+  use CPtr;
+
+  proc recordBatch (args ...?n): GArrowRecordBatch {
+
+    
+    // Verifying the Integrity of the arguments
+    if(n%2==1) then
+      compilerError("Mismatched arguments");
+    for param i in 0..#n {
+      if i%2 == 0 {
+        if args[i].type != string then
+          compilerError("Wrong even argument type");
+      } else {
+        if args[i].type != GArrowArray then
+          compilerError("Wrong odd argument type");
+      }
+    }
```

Comment:
This seems weird, but I have never done anything like this, so I'm not sure if there is a better way. Not the most constructive criticism, but I see why it could be nice to have things done as an array...

Reply:
Unfortunately this is the recommended way of doing this.